src/components/app/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { h, Component } from 'preact';
+import { updater, toggle, When } from '../../lib/util';
+import Fab from '../fab';
+import Header from '../header';
+// import Drawer from 'async!../drawer';
+const Drawer = require('async!../drawer').default;
+import Home from '../home';
+import * as style from './style.scss';
+
+type Props = {
+	url?: String
+}
+
+type FileObj = {
+	id: any,
+	data: any,
+	error: Error | DOMError | String,
+	file: File,
+	loading: Boolean
+};
+
+type State = {
+	showDrawer: Boolean,
+	showFab: Boolean,
+	files: FileObj[]
+};
+
+let counter = 0;
+
+export default class App extends Component<Props, State> {
+	state: State = {
+		showDrawer: false,
+		showFab: true,
+		files: []
+	};
+
+	loadFile = (file: File) => {
+		let fileObj = {
+			id: ++counter,
+			file,
+			error: null,
+			loading: true,
+			data: null
+		};
+
+		this.setState({
+			files: [fileObj]
+		});
+
+		let fr = new FileReader();
+		// fr.readAsArrayBuffer();
+		fr.onerror = () => {
+			let files = this.state.files.slice();
+			files.splice(0, files.indexOf(fileObj), {
+				...fileObj,
+				error: fr.error,
+				loading: false
+			});
+			this.setState({ files });
+		};
+		fr.onloadend = () => {
+			let files = this.state.files.slice();
+			files.splice(0, files.indexOf(fileObj), {
+				...fileObj,
+				data: fr.result,
+				loading: false
+			});
+			this.setState({ files });
+		};
+		fr.readAsDataURL(file);
+	};
+
+	enableDrawer = false;
+
+	openDrawer = updater(this, 'showDrawer', true);
+	closeDrawer = updater(this, 'showDrawer', false);
+	toggleDrawer = updater(this, 'showDrawer', toggle);
+
+	openFab = updater(this, 'showFab', true);
+	closeFab = updater(this, 'showFab', false);
+	toggleFab = updater(this, 'showFab', toggle);
+
+	render({ url }, { showDrawer, showFab, files }) {
+		if (showDrawer) this.enableDrawer = true;
+
+		if (showFab===true) showFab = files.length>0;
+
+		return (
+			<div id="app" class={style.app}>
+				<Fab showing={showFab} />
+
+				<Header toggleDrawer={this.toggleDrawer} loadFile={this.loadFile} />
```

Comment:
Is `toggleDrawe` an event handler? If so, we probably wanna call it `onToggleDrawer` or something?

Reply:
(same for the other eventhandler-y stuff)

Reply:
I assume you mean the left-hand side? If so I agree!

- [ ] `onToggleDrawer`